docs/vc-spec.md

```diff
@@ -1,6 +1,163 @@
+
+Identity provider requests the consent message via `Icrc21ConsentMessageRequest`,
+and upon successful response displays the consent message from `Icrc21ConsentInfo` to the user.
+
+### 2: Prepare Credential
+
+Preparation of a credential involves checking the validity of the request,
+and upon success, preparation of the actual credential requested by the user.
+
+```candid
+service : {
+  prepare_credential : (PrepareCredentialRequest) -> (PrepareCredentialResponse, );
+};
+
+type PrepareCredentialRequest = record {
+    signed_id_alias : SignedIdAlias;
+    credential_spec : CredentialSpec;
+};
+
+type SignedIdAlias = record {
+    credential_jws : text;
+    id_alias : principal;
+    id_dapp : principal;
+};
+
+type PrepareCredentialRequest = record {
+    signed_id_alias : SignedIdAlias;
+    credential_spec : CredentialSpec;
+};
+
+type PrepareCredentialResponse = variant {
+    Ok : PreparedCredentialData;
+    Err : IssueCredentialError;
+};
+
+type PreparedCredentialData = record { vc_jwt : opt text };
+```
+
+Specifically, the issuer checks via `prepared_id_alias.credential_jws` that user identified via `id_dapp` on the issuer
+side can be identified using `id_alias` for the purpose of attribute sharing, and that the credential
+described by `credential_spec` does apply to the user.  When these checks are successful, the issuer
+prepares and returns the credential in JWT format.
+
+**NOTE**: `prepare_credential`-service is needed only for some issuers, e.g. those that use [canister signatures](https://internetcomputer.org/docs/current/references/ic-interface-spec/#canister-signatures)
+during the credential issuance.  For issuers that don't need "preparation"-step, this operation
+can be essentially a no-op that immediately returns `PrepareCredentialResponse::Ok` with an empty
+`PreparedCredentialData.vc_jwt`.
+
+### 3: Get Credential
+
+`get_credential`-service issues the actual credential requested by the user.
+
+```candid
+service : {
+  get_credential : (GetCredentialRequest) -> (GetCredentialResponse) query;
+};
+
+type GetCredentialRequest = record {
+    signed_id_alias : SignedIdAlias;
+    vc_jwt : opt text;
+    credential_spec : CredentialSpec;
+};
+type GetCredentialResponse = variant {
+    Ok : IssuedCredentialData;
+    Err : IssueCredentialError;
+};
+type IssuedCredentialData = record { vc_jws : text };
+
+```
+
+`GetCredentialRequest` should contain the same parameters as `PrepareCredentialRequest`, plus the
+prepared JWT-credential returned by `prepare_credential`, if any.
+The issuer performs the same checks as during the `prepare_credential`-call,
+plus verify that the JWT-credential is consistent with the other parameters.
+
+Upon successful checks, issuer returns the signed credential in JWS-format.
+
+
+
 
-## Window Post Message Interface
+##  Relying Party API (Window Post Message Interface)
```

Comment:
IDP API? (to be consistent with the issuer API being implemented _by_ the issuer)